BKG/Booking.BLL/Services/Booking/BookingService.cs

```diff
@@ -0,0 +1,30 @@
+﻿using AutoMapper;
+using Booking.BLL.Models;
+using Booking.BLL.Services.Booking;
+using Booking.BLL.Services.Generic;
+using Booking.DAL.Entities;
+using Booking.DAL.Repositories.Booking;
+using Booking.DAL.Repositories.Generic;
+
+namespace Booking.BLL.Services.Booking
+{
+    public class BookingService : GenericService<BookingEntity, BookingModel>, IBookingService
+    {
+        private readonly IBookingRepository _bookingRepository;
+
+        public BookingService(IGenericRepository<BookingEntity> repository,
```

Comment:
If you inherit IBookingRepository from IGenericRepository you will avoid this redundant passing of IGenericRepository into base class